scipy/optimize/_lbfgsb_py.py

```diff
@@ -198,7 +198,7 @@ def fmin_l_bfgs_b(func, x0, fprime=None, args=(),
 
     res = _minimize_lbfgsb(fun, x0, args=args, jac=jac, bounds=bounds,
                            **opts)
-    d = {'grad': res['jac'],
+    d = {'grad': res['grad'],
```

Comment:
The whole of `_lbfgsb` has a bunch of `jac` uses that would need to be changed. e.g. `_minimize_lbfgsb` still needs its function sig changed.

Reply:
Would one leave `MemoizeJac` as it is, or rename to `MemoizeGrad`? This might be a function that has a foot in both the grad and jac camps.

Reply:
I ask about this in https://github.com/scipy/scipy/pull/18249#issuecomment-1499210720.
We don't necessarily need to change either of those. This PR would only make the minimum edits to change the public API.